### PR TITLE
#24 certifier can upload ontologies

### PR DESCRIPTION
**Pull request by alvaro-alonso**
this branch is for ticket #24 
It contains the following:
- setup Postgres DB
- setup IPFS
- setup docker-compose
- setup a post route for certifier to post `ontology`
- validate ontology following DTDL guidelines

### CODE REVIEW

File: backend/app/routers/ontologies.py
Comment (amecklenborg) [inline backend/app/routers/ontologies.py:18]: The hash needs still to be defined right? Because this is the dict

File: backend/app/core/ipfs.py
Comment (amecklenborg) [inline backend/app/core/ipfs.py:18]: Good idea to have union types as it has been introduced in the recent python 3.10 version?

File: backend/app/core/ipfs.py
Comment (amecklenborg) [inline backend/app/core/ipfs.py:5]: relative import? The file is on the same directory right?

File: backend/app/routers/ontologies.py
Comment (alvaro-alonso) [reply backend/app/routers/ontologies.py:18]: @amecklenborg the hash is returned by IPFS. It is the unique identifier of record saved.

File: backend/app/core/ipfs.py
Comment (alvaro-alonso) [reply backend/app/core/ipfs.py:18]: @amecklenborg what is the problem here? what would you like to see? I ask because:
- The python version used in the container is 3.9.7 (see the `backend/Dockerfile`) 
-  there is no union type here. Just a type hint that say that variable `ip_addr` is either of type `IPv4Address` or `IPv6Address`

File: backend/app/core/ipfs.py
Comment (alvaro-alonso) [reply backend/app/core/ipfs.py:5]: The `.env` file is not in the same directory, but the package that parse the `.env` is there. See `backend/app/core/config.py`. 

However there should be  a space between `from` and `.config`, I'll fix that.

File: backend/app/core/ipfs.py
Comment (amecklenborg) [reply backend/app/core/ipfs.py:18]: I just say this for avoiding posible compatibility issues. Union types are now allowed in [Python3.10](https://www.python.org/downloads/release/python-3100/)

Comment (alvaro-alonso) [reply backend/app/core/ipfs.py:18]: So you are saying we should update the python version to 3.10 ?

Comment (alvaro-alonso) [reply backend/app/core/ipfs.py:18]: What do you mean with compatibility issues? can you provide an example?

Comment (amecklenborg) [reply backend/app/core/ipfs.py:18]: yes, I am just saying we can update it to 3.10

**Review by amecklenborg — APPROVED** [review]

File: backend/app/core/ipfs.py
Comment (alvaro-alonso) [reply backend/app/core/ipfs.py:18]: But what is the point of updating? Just the union type? the there is a new type in 3.10 doesn't mean that the current syntax won't work in 3.10. If this is just the only reason to update the version is a bit overkill. By changing the version it can happen that others things stop working. I think it is just too risky to change the whole setup version just for a typing class.

**Comment by kaustabhbarman** [comment]
Sorry, I had to setup my system and couldn't review it before it was merged. I will check it out now anyway.

**Comment by alvaro-alonso** [comment]
> Sorry, I had to setup my system and couldn't review it before it was merged. I will check it out now anyway.

sorry I can not merge it today, so I had to do it yesterday. But take a look it will be very helpful for #43. Ping me if you have questions :)

**Pull request merged**
